components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java

```diff
@@ -699,6 +636,95 @@ public String resolveTenantDomain(String organizationId) throws OrganizationMana
         }
     }
 
+    private List<BasicOrganization> getOrganizationsList(boolean allSubOrgs, boolean recursive, Integer limit,
+                                                         String organizationId, String sortOrder,
+                                                         List<ExpressionNode> expressionNodes,
+                                                         List<ExpressionNode> parentIdExpressionNodes)
+            throws OrganizationManagementServerException {
+
+        FilterQueryBuilder filterQueryBuilder = new FilterQueryBuilder();
+        appendFilterQuery(expressionNodes, filterQueryBuilder);
+        Map<String, String> filterAttributeValue = filterQueryBuilder.getFilterAttributeValue();
+        List<String> timestampTypeAttributes = filterQueryBuilder.getTimestampFilterAttributes();
+
+        FilterQueryBuilder parentIdFilterQueryBuilder = new FilterQueryBuilder();
+        appendFilterQueryForParentId(parentIdFilterQueryBuilder, parentIdExpressionNodes);
+        Map<String, String> parentIdFilterAttributeValueMap = parentIdFilterQueryBuilder.getFilterAttributeValue();
+        String parentIdFilterQuery = parentIdFilterQueryBuilder.getFilterQuery();
+
+        String sqlStmt;
+        String getOrgSqlStmtTail = GET_ORGANIZATIONS_TAIL;
+
+        if (isOracleDB()) {
+            getOrgSqlStmtTail = allSubOrgs ? GET_ORGANIZATIONS_TAIL_ORACLE_WITH_ALL_SUB_ORGS :
+                    GET_ORGANIZATIONS_TAIL_ORACLE;
+        } else if (isMSSqlDB()) {
+            getOrgSqlStmtTail = allSubOrgs ? GET_ORGANIZATIONS_TAIL_MSSQL_WITH_ALL_SUB_ORGS
+                    : GET_ORGANIZATIONS_TAIL_MSSQL;
+        }
+
+        if (StringUtils.isBlank(parentIdFilterQuery)) {
+            sqlStmt = GET_ORGANIZATIONS + filterQueryBuilder.getFilterQuery() +
+                    String.format(getOrgSqlStmtTail, SET_ID, recursive ? "> 0" : "= 1", sortOrder);
+        } else {
+            sqlStmt = GET_ORGANIZATIONS + filterQueryBuilder.getFilterQuery() +
+                    String.format(getOrgSqlStmtTail, parentIdFilterQuery, recursive ? "> 0" : "= 1",
+                            sortOrder);
+        }
+
+        String permissionPlaceholder = "PERMISSION_";
+        List<String> permissions = getAllowedPermissions(VIEW_ORGANIZATION_PERMISSION);
+        List<String> permissionPlaceholders = new ArrayList<>();
+        // Constructing the placeholders required to hold the permission strings in the named prepared statement.
+        for (int i = 1; i <= permissions.size(); i++) {
+            permissionPlaceholders.add(":" + permissionPlaceholder + i + ";");
+        }
+        String placeholder = String.join(", ", permissionPlaceholders);
+        sqlStmt = sqlStmt.replace(PERMISSION_LIST_PLACEHOLDER, placeholder);
```

Comment:
is this required if allSubOrgs == true ?

Reply:
we can stop them as well